src/controllers/reviewController.ts

```diff
@@ -0,0 +1,74 @@
+import { Request, Response } from "express";
+import Review from "../models/Review";
+
+export const createReview = async (req: Request, res: Response) => {
+    const { content, postId, parentReview } = req.body;
+    const userId = req.user.userId;
+
+    if (!userId) {
+        return res.status(401).send({ error: 'Unauthorized' });
+    }
+
+    try {
+        const review = new Review({
+            userId,
+            postId,
+            content,
+            parentReview
+        });
+
+        if (parentReview) {
+            const parent = await Review.findById(parentReview);
+            if (parent) {
+                parent.childrenReview.push(review._id);
+                await parent.save();
+            }
+        }
+
+        await review.save();
+        res.status(201).send(review);
+    }
+    catch (error) {
+        res.status(400).send(error);
+    }
+};
+
+export const updateReview = async (req: Request, res: Response) => {
+    const { id } = req.params;
+    const { content } = req.body;
+    const userId = req.user.userId;
+
+    if (!userId) {
+        return res.status(401).send({ error: 'Unauthorized' });
+    }
+
+    try {
+        const review = await Review.findByIdAndUpdate(id, 
+            { content }, 
+            { new: true, runValidators: true });
+        if (review?.userId.toString() !== userId) {
+            return res.status(403).send({ error: 'Forbidden' });
+        }
+
+        review.content = content;
+        await review.save();
+
+        res.status(200).send(review);
+    }
+    catch (error) {
+        res.status(400).send(error);
+    }
+
+};
+
+export const deleteReview = async (req: Request, res: Response) => {
```

Comment:
You should also access the post id through req.params and then remove this review id from the post array. Also remember to delete it from the parent review array.